### PR TITLE
Improve log with renames

### PR DESCRIPTION
**Pull request by dometto**
### With list_renames => true

```
Rehearsal ----------------------------------------------
             3.540000   0.500000   4.040000 (  3.763362)
------------------------------------- total: 4.040000sec

                 user     system      total        real
             3.120000   0.440000   3.560000 (  3.522378)
```

### With log_renames => false

```
Rehearsal ----------------------------------------------
             2.380000   0.310000   2.690000 (  2.389807)
------------------------------------- total: 2.690000sec

                 user     system      total        real
             1.850000   0.240000   2.090000 (  2.015580)
```

### CODE REVIEW

**Comment by dometto** [comment]
After refactoring, this performs better than the approach in #52, and is actually accurate.

@bartkamphorst: what do you think of the approach of adding `Commit#tracked_pathname`? On the one hand, it feels a bit odd to add info about a single pathname to a commit. On the other hand, I think the use case 'follow rename on a single file' is legit, and this seems a fairly user-friendly way of dealing with it.

**Comment by dometto** [comment]
@bartkamphorst I went with an option you suggested elsewhere, which is to add the tracked_pathname attribute to a separate subclass of Commit (now called TrackingCommit). I think this offers a pretty clean implementation. If you agree, we can merge this and cut a new RJGit release!

File: lib/git.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def log(path = nil, revstring = Constants::HEAD, options = {})
       if path && options[:follow]
         cfg       = Configuration.new(nil)
         cfg.add_setting('renames', true, 'diffs', nil)
-        follow    = FollowFilter.create(path, cfg.get(org.eclipse.jgit.diff.DiffConfig::KEY))
+        follow    = FollowFilter.create(path, cfg.jconfig.get(org.eclipse.jgit.diff.DiffConfig::KEY))
```
Comment (bartkamphorst) [inline lib/git.rb]: Can you explain this change? Did `cfg.get` work before or not at all?

Comment (dometto) [reply lib/git.rb]: It worked when running the specs on RJGit, but not when trying to call this method from the rjgit-adapter. My guess is that context somehow breaks the method_missing magic that this relied on, so perhaps better to be explicit.

Comment (bartkamphorst) [reply lib/git.rb]: Okay!

File: lib/commit.rb [hunk 2]
```diff
@@ -130,4 +128,13 @@ def self.find_all(repository, ref, options)
     end
   
   end
+  
+  class TrackingCommit < Commit
```
Comment (bartkamphorst) [inline lib/commit.rb]: Very clean indeed! Nice. 👍

Comment (dometto) [reply lib/commit.rb]: Thanks!

**Pull request merged**